### PR TITLE
Fixes #90 by configuring webhook not to get called on status updates.

### LINKED ISSUE #90: Do not validate / default on /status updates.
**Description**

As part of the #88 I added a skip check for status. Once https://github.com/knative/pkg/issues/2115 gets fixed, we should configure the policy-controller to completely opt out of the status path.

### PR DESCRIPTION
**Pull request by vaikas**
Signed-off-by: Ville Aikas <vaikas@chainguard.dev>

This fixes #90 by not unnecessarily getting invoked for status updates (which we handle but it's noisy in the logs) as well as for deletes. Until I get this wrangled into the upstream, this is just showing using my personal branch.

After this change, the new reconciled validatingwebhookconfiguration looks like this:
```
  rules:
  - apiGroups:
    - ""
    apiVersions:
    - v1
    operations:
    - CREATE
    - UPDATE
    resources:
    - pods
    scope: '*'
  - apiGroups:
    - apps
    apiVersions:
    - v1
    operations:
    - CREATE
    - UPDATE
    resources:
    - daemonsets
    scope: '*'
  - apiGroups:
    - apps
    apiVersions:
    - v1
    operations:
    - CREATE
    - UPDATE
    resources:
    - deployments
    scope: '*'
```

Compared to previous:
```
  rules:
  - apiGroups:
    - ""
    apiVersions:
    - v1
    operations:
    - CREATE
    - UPDATE
    - DELETE
    resources:
    - pods
    - pods/status
    scope: '*'
  - apiGroups:
    - apps
    apiVersions:
    - v1
    operations:
    - CREATE
    - UPDATE
    - DELETE
    resources:
    - daemonsets
    - daemonsets/status
    scope: '*'
  - apiGroups:
    - apps
    apiVersions:
    - v1
    operations:
    - CREATE
    - UPDATE
    - DELETE
    resources:
    - deployments
    - deployments/status
    scope: '*'
```

### CODE REVIEW

File: cmd/webhook/main.go [hunk 1]
```diff
@@ -98,17 +99,37 @@ func main() {
...
+}
+
+func (c *crdNoStatusUpdatesOrDeletes) SupportedSubResources() []string {
+	// We do not want any updates that are for status, scale, or anything else.
+	return []string{""}
+}
+
+func (c *crdNoStatusUpdatesOrDeletes) SupportedVerbs() []admissionregistrationv1.OperationType {
+	return []admissionregistrationv1.OperationType{admissionregistrationv1.Create,
+		admissionregistrationv1.Update,
```
Comment (mattmoor) [inline cmd/webhook/main.go:118]: Can you move Create onto its own line as well?

**Review by mattmoor — APPROVED** [review]
Any good way to test this?

File: cmd/webhook/main.go (hunk 1 above)
Comment (vaikas) [reply cmd/webhook/main.go]: nope

**Comment by vaikas** [comment]
Hm. Maybe write an e2e test that checks the validatingwebhookconfiguration to make sure there's no status subresources. Or maybe if we have metrics to see how many times the webhook was called. Let's think about it. Open to ideas :)

**Comment by mattmoor** [comment]
LOL, that's better than what I was thinking!

**Review by mattmoor — APPROVED** [review]

**Review by cpanato — APPROVED** [review]

**Pull request merged**